### PR TITLE
Story Data: Single Animation Target

### LINKED ISSUE #4153: Update Animations so Targets contain just one Element ID
Originally we coded group animations so each element in the group would have the exact same animation properties (i.e. same duration, delay, etc..), however, after some design thought, it's been concluded that even grouped elements should have the ability to have separate animation properties.  This means that our existing logic/template animations must only have one target element.

**Things to do**
1.)  Update all raw Template files to break out animations with multiple target elements into multiple animation records (perhaps with a quick one time script).
2.)  Update the `story-anim-tool` so when animations are applied to a group of elements, multiple animation records are created and not an array of IDs in `targets`.

### PR DESCRIPTION
**Pull request by maxyinger**
## Summary
- Used node script to update exisiting template data animation data to be single target
- Updated story-anim-tool to apply a separate animation to each selected target
- Added migration script to make any existing stories created from outdated templates to only have a single target per animation

## Relevant Technical Choices
used node script to update existing template data, then realized this step should probably be part of the migration step cus ppl may have stories created from the old template data.

## To-do
Seems there was a regression in the story anim tool where the story animation data updates no longer hit the story api. It appears `updateStory` was altered recently in the dashboard to only be able to update the story title.

This regression feels a little separate from this ticket, I'll create a separate PR to fix it. Not sure how much we'll be using the story-anim-tool from now on to actually edit story animations anyways 🤷 

## User-facing changes
None

## Testing Instructions
Enable animations and see that they still play & work in all the templates and on stories with animations applied.

---

Fixes #4153

### CODE REVIEW

**Review by BrittanyIRL — COMMENTED** [review]
When I select "use template" on a template in the dashboard and click on an element, the animations that are built into the template no longer show their title in the right side design panel (see screenshot), but when i add another animation (like "drop" here) it does show in the sidebar. I'm not getting any errors or warnings. 

Screen Shot 2020-11-05 at 10 48 09 AM
Screen Shot 2020-11-05 at 10 48 32 AM

**Comment by maxyinger** [comment]
> When I select "use template" on a template in the dashboard and click on an element, the animations that are built into the template no longer show their title in the right side design panel (see screenshot), but when i add another animation (like "drop" here) it does show in the sidebar. I'm not getting any errors or warnings.
> 
> Screen Shot 2020-11-05 at 10 48 09 AM
> 
> Screen Shot 2020-11-05 at 10 48 32 AM

Ah yea, thats cus the templates use animation parts, not the animation effects. This PR I currently have open should take care of that:
https://github.com/google/web-stories-wp/pull/4906

[just merged that in, gonna rebase this PR now & should be good]

**Review by BrittanyIRL — APPROVED** [review]
Ah, sorry about that! I thought that was already merged. This looks great 👍

File: assets/src/dashboard/app/views/storyAnimTool/index.js [hunk 1]
```diff
@@ -161,27 +162,32 @@ function StoryAnimTool() {
...
+
+          if (index < 0) {
+            // not found, push it to the end
+            animations.push(animationWithTarget);
+          } else {
+            // found, overwrite at index
+            animations[index] = animationWithTarget;
+          }
+
+          story.pages[activePageIndex].animations = [...animations];
```
Comment (carloskelly13) [inline assets/src/dashboard/app/views/storyAnimTool/index.js]: Do we want a deep copy here or just a new array with elements?

**Review by carloskelly13 — APPROVED** [review]
Looks good with 1 question

File: assets/src/dashboard/app/views/storyAnimTool/index.js (hunk 1 above)
Comment (maxyinger) [reply assets/src/dashboard/app/views/storyAnimTool/index.js]: I honestly don't even see the need for all this copying back n forth and think it's fine to operate on the accumulator directly. I was just trying to change as little as possible and this was in the old code

Happy to refactor & clean up a bit

Also mind clarifying on deep copy vs new array with elements? don't think I'm 100% following you there

Comment (embarks) [reply assets/src/dashboard/app/views/storyAnimTool/index.js]: I don't _think_ we need a deep copy here, and should be more performant without one

**Review by embarks — APPROVED** [review]

**Pull request merged**